nemo/collections/asr/parts/submodules/cuda_graph_rnnt_greedy_decoding.py

```diff
@@ -104,96 +72,7 @@ def create_inner_while_loop_kernel():
-
-    cuda.cuLaunchKernel(
-        while_loop_kernel,
-        1,
-        1,
-        1,
-        1,
-        1,
-        1,
-        0,
-        torch.cuda.current_stream(device=device).cuda_stream,
-        while_loop_args.ctypes.data,
-        0,
-    )
-
-    capture_status, _, graph, dependencies, _ = cu_call(
-        cudart.cudaStreamGetCaptureInfo(torch.cuda.current_stream(device=device).cuda_stream)
-    )
-    assert capture_status == cudart.cudaStreamCaptureStatus.cudaStreamCaptureStatusActive
-
-    driver_params = cuda.CUgraphNodeParams()
-    driver_params.type = cuda.CUgraphNodeType.CU_GRAPH_NODE_TYPE_CONDITIONAL
-    driver_params.conditional.handle = while_loop_conditional_handle
-    driver_params.conditional.type = cuda.CUgraphConditionalNodeType.CU_GRAPH_COND_TYPE_WHILE
-    driver_params.conditional.size = 1
-    if Version(cuda_python_version) == Version("12.3.0"):
-        # Work around for https://github.com/NVIDIA/cuda-python/issues/55
-        # Originally, cuda-python version 12.3.0 failed to allocate phGraph_out
-        # on its own.
-        # This bug is fixed in cuda-python version 12.4.0. In fact, we can
-        # no longer write to phGraph_out in cuda-python 12.4.0, so we must
-        # condition on the version number.
-        driver_params.conditional.phGraph_out = [cuda.CUgraph()]
-    (ctx,) = cu_call(cuda.cuCtxGetCurrent())
-    driver_params.conditional.ctx = ctx
-
-    # Use driver API here because of bug in cuda-python runtime API: https://github.com/NVIDIA/cuda-python/issues/55
-    # TODO: Change call to this after fix goes in (and we bump minimum cuda-python version to 12.4.0):
-    # node, = cu_call(cudart.cudaGraphAddNode(graph, dependencies, len(dependencies), driver_params))
-    (node,) = cu_call(cuda.cuGraphAddNode(graph, dependencies, len(dependencies), driver_params))
-    body_graph = driver_params.conditional.phGraph_out[0]
-
-    cu_call(
-        cudart.cudaStreamUpdateCaptureDependencies(
-            torch.cuda.current_stream(device=device).cuda_stream,
-            [node],
-            1,
-            cudart.cudaStreamUpdateCaptureDependenciesFlags.cudaStreamSetCaptureDependencies,
-        )
-    )
-    body_stream = torch.cuda.Stream(device)
-    previous_stream = torch.cuda.current_stream(device=device)
-    cu_call(
-        cudart.cudaStreamBeginCaptureToGraph(
-            body_stream.cuda_stream,
-            body_graph,
-            None,
-            None,
-            0,
-            cudart.cudaStreamCaptureMode.cudaStreamCaptureModeThreadLocal,
-        )
-    )
-    torch.cuda.set_stream(body_stream)
-
-    yield body_stream, body_graph
-
-    cuda.cuLaunchKernel(
-        while_loop_kernel, 1, 1, 1, 1, 1, 1, 0, body_stream.cuda_stream, while_loop_args.ctypes.data, 0
-    )
-
-    cudart.cudaStreamEndCapture(body_stream.cuda_stream)
-
-    torch.cuda.set_stream(previous_stream)
+    return run_nvrtc(kernel_string, b"while_loop_conditional", _CUDA_PROGRAM_NAME)
```

Comment:
cuda-python expects a byte string, not a normal python string. It is because cuda expects a `char *` in the underlying C API. Just a silly implementation detail.